django/core/paginator.py

```diff
@@ -83,13 +84,17 @@ def _get_page(self, *args, **kwargs):
     @cached_property
     def count(self):
         """Return the total number of objects, across all pages."""
+        object_list = self.object_list
```

Comment:
i don't know what the official Django community's take on ternary operators or `or` assignment is, but you could consolidate these three lines into:
`object_list = self.count_object_list if self.count_object_list else self.object_list`, 
or better yet:
`object_list = self.count_object_list or self.object_list`